### PR TITLE
EnC support for DIM

### PR DESCRIPTION
**Pull request by tmat**
Allow editing existing interface member bodies.

Disallow adding new ones - virtual and abstract ones are not supported in general in abstract types, non-virtual should in theory work but don't (https://github.com/dotnet/roslyn/issues/37128).

Adding a nested types is supported.

Disallow declaring a new local function in a body of an interface method. We synthesize the IL method directly to the containing interface, which does not work. Lambdas, async, and iterators are supported since they synthesize a nested type.

### CODE REVIEW

**Comment by tmat** [comment]
@ivanbasov @jcouv PTAL
No changes in the compiler, only tests added.

File: src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs [hunk 1]
```diff
@@ -2009,20 +2008,37 @@ private void ClassifyInsert(SyntaxNode node)
                 }
             }
 
-            private bool ClassifyModifiedMemberInsert(SyntaxTokenList modifiers)
+            private bool ClassifyModifiedMemberInsert(MemberDeclarationSyntax declaration, SyntaxTokenList modifiers)
```
Comment (ivanbasov) [inline src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs]: >MemberDeclarationSyntax declaration, SyntaxTokenList modifiers [](start = 54, length = 62)

Should we get modifiers from declaration.Modifiers?

**Comment by tmat** [comment]
@agocke @jcouv PTAL -- no compiler changes, just test changes

File: src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTests.cs [hunk 2]
```diff
@@ -1936,6 +1936,202 @@ public C()
...
+            var reader2 = md2.Reader;
+            readers = new[] { reader0, reader1, reader2 };
+
+            CheckNames(readers, reader2.GetTypeDefNames());
+            CheckNames(readers, reader2.GetFieldDefNames(), "X");
+            CheckNames(readers, reader2.GetMethodDefNames(), "M", "N", "get_P", "set_P", "get_Q", "set_Q", "add_E", "remove_E", "add_F", "remove_F", ".cctor");
+            Assert.Equal(0, reader2.GetTableRowCount(TableIndex.NestedClass));
+
+            CheckEncLog(reader2,
+                Row(3, TableIndex.AssemblyRef, EditAndContinueOperation.Default),
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTests.cs]: >Row(3, [](start = 16, length = 6)

I don't know how to interpret these numbers. Are they important?

Comment (tmat) [reply src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTests.cs]: These are row ids. Each line here corresponds to an entry in EnC log that says what operation was done on what entity (row number + table id). The specific values are not that interesting. The interesting part is that we are emitting a couple of events, a field, a bunch of MethodDefs and and a couple of properties.

**Pull request merged**